src/types/mailbox.rs

```diff
@@ -35,6 +35,10 @@ pub struct Mailbox {
     /// The unique identifier validity value.  See [`Uid`] for more details.  If this is missing,
     /// the server does not support unique identifiers.
     pub uid_validity: Option<u32>,
+
+    /// The highest mod sequence for this mailboxr. Used with
+    /// [Conditional STORE](https://tools.ietf.org/html/rfc4551#section-3.1.1).
+    pub highest_mod_seq: Option<u64>,
```

Comment:
We should probably add `#[non_exhaustive]` to `Mailbox` as well so that we can make these kinds of changes in the future without a breaking change.

Reply:
Ah, that is a good idea! I added this change.